### DISCUSSION TITLE
preloaded images are loaded twice

### DISCUSSION THREAD

**Discussion by fhaselon** [jspsych/jsPsych#1396] [Q&A]
Hello everyone,

I am new to jspsych and I am writing an experiment that will run on a smaller JATOS-server. As it includes a reaction-time task, I would like the experiment to preload all the stimulus files (.jpgs). Currently, my browser creates the imagefolder and preloads all the images into that folder before starting the experiment, by using the preload-function. However, while running the experiment I recognized, that every imagefile, that is displayed during runtime, is loaded a second time and stored in the folder. When I turn of the internet connection while running the experiment, the task crashes. I hosted the task from localhost and Pavlovia to check whether this effects the behavior.

In brief, my experiment preloads all the stimuli but does not use them. Instead, it loads them again when needed. This happens when I specify the images that should be preloaded as well as when I leave this option to auto.  

Does anyone have an idea how to solve this problem? 

Thanks for any help!

**Comment by jodeleeuw** [comment]
Hi @fhaselon,

Do you have you cache disabled for the page or in general? Unfortunately there's only so much that JavaScript can control here, so if a user's settings prohibit caching then we can't preload anything.
